core/state_transition.go

```diff
@@ -337,7 +337,14 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 		effectiveTip = cmath.BigMin(st.gasTipCap, new(big.Int).Sub(st.gasFeeCap, st.evm.Context.BaseFee))
 	}
 	amount := new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), effectiveTip)
+	if london {
+		burntContractAddress := common.HexToAddress(st.evm.ChainConfig().Bor.CalculateBurntContract(st.evm.Context.BlockNumber.Uint64()))
+		burnAmount := new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), st.evm.Context.BaseFee)
+		st.state.AddBalance(burntContractAddress, burnAmount)
```

Comment:
Not sure how comfortable I am with this logic here so different from Geth. Have we though about different ways to do this?

Reply:
we are just sending the burnAmount to one of the contracts in the genesis file which in geth just burns

Reply:
But how do we know what the burn amount is? There are no test. IMHO, I really do not understand what are we doing here. Do we have any documents explaining it?

Reply:
@prabal-banerjee can you confirm the burn logic

Reply:
Logic seems to be correct, but I agree with Ferran about writing test cases to verify correct amount is burnt and correct amount is transferred to miner.